src/types/ScreenInfoUiaProviderBase.cpp

```diff
@@ -38,30 +40,34 @@ try
     RETURN_HR_IF_NULL(E_INVALIDARG, pData);
     _pData = pData;
     _wordDelimiters = wordDelimiters;
+
+    _id = id;
+    ++id;

```

Comment:
Should this be done atomically? idk

Reply:
I was having trouble coming up with a way to do it automatically but I think I just thought of one.

Plan:
- keep a count for SIUP and UTRs in UiaTracing
- when a constructor is traced, assign an ID based on the count above

It'd make constructors non-const but I guess that'd be ok?

Reply:
Oh I didn't typo there, I meant like `atomic`-ly, so if two are constructed on different threads they don't each get the same ID

Reply:
oh man. I completely misread that haha. I've never used `atomic` before, but I'll add it in just in case.

Reply:
I wouldn't overcomplicate this. This an ID used only for UIA tracing, UIA calls only come in on one thread or so, _and_ it's not dangerous to the application in the least if a value is reused or skipped. No compelling need to make it atomic.

Reply:
But one could fold this into one line, `_id = id++`

Reply:
outdated now so resolving.